test/array_serializer_test.rb

```diff
@@ -39,15 +45,52 @@ def test_meta_and_meta_key_attr_readers
         assert_equal @serializer.meta_key, "the meta key"
       end
 
-      def test_json_key_when_resource_is_empty
```

Comment:
Maybe I'm being slow and missing your point, in the `build_named_collection` helper method, it's returning an instance with a name of `"MeResource"` which gets underscored & pluralized to `"me_resources"`. 

If I understand #1007 correctly, it was a problem with an empty collection object with a `name` not getting transformed properly. I believe that `test_json_key_with_resource_with_name_and_no_serializers` is effectively testing that case, right?